pyanaconda/modules/users/installation.py

```diff
@@ -0,0 +1,159 @@
+# modify, copy, or redistribute it subject to the terms and conditions of
+# the GNU General Public License v.2, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY expressed or implied, including the implied warranties of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+
+from pyanaconda.core import users
+
+from pyanaconda.modules.common.task import Task
+
+from pyanaconda.anaconda_loggers import get_module_logger
+log = get_module_logger(__name__)
+
+__all__ = ["SetRootPasswordTask", "CreateUsersTask", "CreateGroupsTask", "SetSshKeysTask"]
+
+
+class SetRootPasswordTask(Task):
+    """Installation task fot setting the root password."""
+
+    def __init__(self, sysroot, password, crypted, locked):
+        """Create a new root password configuration task.
+
+        :param str sysroot: a path to the root of the target system
+        :param str password: root password to be set
+        :param bool crypted: is the root password is already encrypted ?
+        :param bool locked: should the root password be locked ?
+        """
+        super().__init__()
+        self._sysroot = sysroot
+        self._password = password
+        self._crypted = crypted
+        self._locked = locked
+
+    @property
+    def name(self):
+        return "Configure root password"
+
+    def run(self):
+        self._set_root_password()
+
+    def _set_root_password(self):
+        users.set_root_password(self._password,
+                              self._crypted,
+                              self._locked,
+                              self._sysroot)
+
+
+class CreateUsersTask(Task):
+    """Create users on the target system."""
+
+    def __init__(self, sysroot, user_data_list):
+        """Create a new user creation task.
+
+        :param str sysroot: a path to the root of the installed system
+        :param user_data_list: list of users to create
+        :type user_data_list: list of UserData instances
+        """
+        super().__init__()
+        self._sysroot = sysroot
+        self._user_data_list = user_data_list
+
+    @property
+    def name(self):
+        return "Create users"
+
+    def run(self):
+        self._create_users()
+
+    def _create_users(self):
+        for user_data in self._user_data_list:
+            # UserData uses -1 for not-set uid/gid while the function takes None for not-set
```

Comment:
Let's rather call it `USER_UID_NOT_SET`, `USER_GID_NOT_SET` & `GROUP_GID_NOT_SET` & put it next to the data classes.

Reply:
`USER_UID_NOT_SET` suggests that the UID doesn't have a valid value, so it will not be possible to create the user until it is set. However, we will create a default UID. I would like to keep it simple. Also, do we really need different constants for user GID and group GID?

Reply:
I think that it this cpntext it's about the value being defined as not set int  the specification requested for the tigen user or group. `DEFAULT_UID` or `DEFAULT_GID` would imply Anaconda would try to create a user with a UID/GID equal to -1, which is wrong and likely impossible.

As for different user and group GID - well, that basically due to where the constants are defined, one in the module holding the `UserData` class, one in the module holding the `GroupData` class. 

Potentially, there could be just a single "not set" constant defined, but naming it is tricky, as it's basically using a -1 to indciate that a value that can't be <0 has not been set. Maybe `POSITIVE_VALUE_NOT_SET` ? That might be generic enough to go to the global contants file.

In any case I don't think this is that important to require and imrovement in this PR and can be added as a separate PR later on.

Reply:
Yes, it doesn't need to be fixed now. We can discuss it later.